docs/src/app/(private)/experiments/menu/complex-nesting.tsx

```diff
@@ -48,7 +48,7 @@ export default function MenuComplexNestingExperiment() {
 
                 <Dialog.Root>
                   <Menu.Item
-                    render={<Dialog.Trigger />}
+                    render={<Dialog.Trigger nativeButton={false} />}
```

Comment:
This doesn't seem correct. Dialog.Trigger renders a native button. If there was `<Dialog.Trigger render={<Menu.Item />}>` we'd need `nativeButton={false}`.

Reply:
Yes, right - imo this area needs work in the API as it's pretty confusing and an easy footgun

Reply:
It could be alleviated by us using native buttons everywhere, including menu items. Then, only if devs want to use noninteractive elements, they'll have to explicitly add the prop.